compiler/sem/pragmas.nim

```diff
@@ -1563,90 +1377,56 @@ proc prepareSinglePragma(
       of wRequiresInit:
         result = noVal(c, it)
         if sym.kind == skField:
           sym.flags.incl sfRequiresInit
-        elif sym.typ != nil:
-          incl(sym.typ.flags, tfNeedsFullInit)
         else:
-          result = invalidPragma(c, it)
+          assert sym.typ != nil
+          incl(sym.typ.flags, tfNeedsFullInit)
       of wByRef:
         result = noVal(c, it)
-        if sym == nil or sym.typ == nil:
-          var tmp = c.config.options
-          result = processOption(c, it, tmp)
-          c.config.options = tmp
-        else:
-          incl(sym.typ.flags, tfByRef)
+        assert sym != nil and sym.typ != nil
+        incl(sym.typ.flags, tfByRef)
       of wByCopy:
         result = noVal(c, it)
-        if sym.kind != skType or sym.typ == nil:
-          result = invalidPragma(c, it)
-        else:
-          incl(sym.typ.flags, tfByCopy)
+        assert sym.kind == skType and sym.typ != nil
+        incl(sym.typ.flags, tfByCopy)
       of wInject, wGensym:
         # We check for errors, but do nothing with these pragmas otherwise
         # as they are handled directly in 'evalTemplate'.
         result = noVal(c, it)
-        if sym == nil: result = invalidPragma(c, it)
-      of wLine:
-        result = pragmaLine(c, it)
       of wRaises, wTags:
         result = pragmaRaisesOrTags(c, it)
       of wLocks:
-        if sym == nil:
-          result = pragmaLockStmt(c, it)
-        elif sym.typ == nil:
-          result = invalidPragma(c, it)
-        else:
           (sym.typ.lockLevel, result) = pragmaLocks(c, it)
           if result.isNil:
             result = it
       of wBitsize:
-        if sym == nil or sym.kind != skField:
-          result = invalidPragma(c, it)
-        else:
+          assert sym.kind == skField
           (sym.bitsize, result) = intLitToIntOrErr(c, it)
           if result.isNil: result = it
           if sym.bitsize <= 0:
             result = c.config.newError(it,
                       PAstDiag(kind: adSemBitsizeRequiresPositive))
       of wGuard:
-        result = it
-        if sym == nil or sym.kind notin {skVar, skLet, skField}:
-          result = invalidPragma(c, it)
-        else:
+          assert sym.kind in {skVar, skLet, skField}
           sym.guard = pragmaGuard(c, it, sym.kind)
           if sym.guard != nil and sym.guard.kind == skError and sym.guard.ast != nil and sym.guard.ast.kind == nkError:
             result = sym.guard.ast
           else:
             result = it
       of wGoto:
-        result = it
-        if sym == nil or sym.kind notin {skVar, skLet}:
-          result = invalidPragma(c, it)
-        else:
-          sym.flags.incl sfGoto
+        result = noVal(c, it)
+        assert sym.kind in {skVar, skLet}
+        sym.flags.incl sfGoto
       of wExportNims:
-        result = it
-        if sym == nil:
-          result = invalidPragma(c, it)
-        else:
+          # XXX: modifying the module graph during application of a symbol
```

Comment:
At first I thought it was in effect export it, but nope, it's adding it to `exposed`, which I believe amounts to export it at the codegen level, but not the semantic level. I can't seem to find any usage of `exportnims` either -- odd.

Reply:
I believe it's meant for registering a procedure to later be able to look it up by name, when embedding NimScript.

Reply:
Yup, at the time I made the comment I did some searches trying to find the mechanism where things were actually stitched together. It was really difficult to find and couldn't get a complete picture. With that said, if that specific bit is revisited, that whole mechanism likely needs to be revised and perhaps this pragma will be shown to be unnecessary.